tests/modal-v2/modal-v2.spec.tsx

```diff
@@ -0,0 +1,92 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import { ModalV2 } from "src/modal-v2";
+
+// =============================================================================
+// UNIT TESTS
+// =============================================================================
+describe("ModalV2", () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it("should render the component", () => {
+        render(
+            <ModalV2 show>
+                <ModalV2.Card>
+                    <ModalV2.CloseButton />
+                    <ModalV2.Content>
+                        <div data-testid="test-content" />
+                    </ModalV2.Content>
+                    <ModalV2.Footer
+                        primaryButton={<button data-testid="test-button" />}
+                    />
+                </ModalV2.Card>
+            </ModalV2>
+        );
+
+        // assert slots are rendered
+        expect(screen.getByTestId("modal-close-button")).toBeInTheDocument();
+        expect(screen.getByTestId("modal-content")).toBeInTheDocument();
+        expect(screen.getByTestId("modal-footer")).toBeInTheDocument();
+
+        // assert content of slots are rendered
+        expect(screen.getByTestId("test-content")).toBeInTheDocument();
+        expect(screen.getByTestId("test-button")).toBeInTheDocument();
+    });
+
+    it("should not render unknown slots in Modal.Card", () => {
+        render(
+            <ModalV2 show>
```

Comment:
just showing off being pedantic but don't necessarily need the modal for this test